[Workplace AI] Gmail Data Source (#255565)
elastic/kibana: docs/reference/connectors-kibana/gmail-action-type.md

---
navigation_title: "Gmail"
type: reference

> description: "Use the Gmail connector to search and read emails from Gmail in Workplace AI and Agent Builder."

Reviewer: ```suggestion
description: "Use the Gmail connector to search and read emails from Gmail."
```
I don't think we should mention workplace AI

applies_to:
  stack: preview 9.4
  serverless: preview
---

# Gmail connector [gmail-action-type]

> The Gmail connector connects to the Gmail API and enables federated search of email in Workplace AI and Agent Builder. You configure a Bearer token (Google OAuth 2.0 access token) when creating the connector.

Reviewer: ```suggestion
The Gmail connector connects to the Gmail API and enables federated search of email. You configure a Bearer token (Google OAuth 2.0 access token) when creating the connector.
```

## Create connectors in {{kib}} [define-gmail-ui]

You can create a Gmail connector in **{{stack-manage-app}} > {{connectors-ui}}** or when adding a Gmail data source.

### Connector configuration [gmail-connector-configuration]

Gmail connectors use the following configuration:

Bearer Token
:   A Google OAuth 2.0 access token with Gmail API scopes. See [Get API credentials](#gmail-api-credentials) for instructions.

## Test connectors [gmail-action-configuration]

You can test connectors when creating or editing the connector in {{kib}}. The test verifies connectivity by fetching the authenticated user's profile from the Gmail API.

## Available actions [gmail-available-actions]

| Action | Description |
|--------|-------------|
| Search messages | Search for messages using Gmail search syntax. Parameters: `query` (optional), `maxResults` (optional, default 10, max 100), `pageToken` (optional). |

> | List messages | List message IDs, optionally filtered by label. Parameters: `maxResults` (optional, default 10, max 100), `pageToken` (optional), `labelIds` (optional, e.g. INBOX, SENT). |

Reviewer: ```suggestion
| List messages | List message IDs, optionally filtered by label. Parameters: `maxResults` (optional, default 10, max 100), `pageToken` (optional), `labelIds` (optional, for example: INBOX, SENT). |
```
